### PR TITLE
Add org.gnome.NetworkDisplays

### PR DESCRIPTION
**Pull request by Lctrs**

### CODE REVIEW

**Comment by Lctrs** [comment]
Friendly ping @benzea

**Comment by nedrichards** [comment]
bot, build org.gnome.NetworkDisplays

**Comment by benzea** [comment]
Ah, didn't see this yet.

My main concern here would be having the appropriate gstreamer elements. i.e. we also need the `openh264enc`, `vaapih264enc` or `x264enc` and for audio `fdkaacenc` or `avenc_aac`. I think there is a bundle containing vaapienc at least, so that might be the best option.

Note that those are purely runtime dependencies, they are not required for building.

**Comment by Lctrs** [comment]
@benzea `openh264enc` and `fdkaacenc` are available in the runtime.

**Comment by nedrichards** [comment]
bot, build org.gnome.NetworkDisplays

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "publish-delay-hours": 1
```
Comment (barthalion) [inline flathub.json:2]: Any reason to change it from the default?

Comment (Lctrs) [reply flathub.json:2]: Just a personal preference.

**Comment by nedrichards** [comment]
bot, build org.gnome.NetworkDisplays

**Comment by nedrichards** [comment]
I can't test this artifact on my machine (inappropriate hardware and distro setup) so will need confirmation that the thing actually does what it's supposed to.

**Comment by benzea** [comment]
```
 1. [✓] org.gnome.Platform                   x86_64 3.36   i  flathub                  153.0 MB / 324.5 MB
 2. [✓] org.gnome.Platform.Locale            x86_64 3.36   i  flathub                   16.8 kB / 323.0 MB
 3. [✓] org.freedesktop.Platform.GL.default  x86_64 19.08  u  flathub                   84.9 MB / 85.1 MB
 4. [✓] org.freedesktop.Platform.VAAPI.Intel x86_64 19.08  u  flathub                    1.1 kB / 8.7 MB
 5. [✗] org.freedesktop.Platform.openh264    x86_64 2.0    u  gnome-nightly          < 916.8 kB
 6. [✓] org.gnome.NetworkDisplays            x86_64 test   i  networkdisplays-origin     7.3 MB / 7.7 MB
 7. [✓] org.gnome.NetworkDisplays.Locale     x86_64 test   i  networkdisplays-origin     5.1 kB / 2.3 MB
```

So, I think in theory it works, however I am getting:

```
[E][connection.c:513 pw_protocol_native_connection_flush()] could not sendmsg: Destination address required
```

Which I suspect could be a pipewire version conflict (on F31 right now, which is pipewire 0.2.7 rather than 0.3)

**Comment by benzea** [comment]
Well, on F32 and a new laptop I now have `wpa_supplicant` crash on me. Which may be related to the device name getting too long or something.

Either way, it might not be a good idea if you sink a lot of time into this. I fear there are some things broken currently :)

**Comment by Lctrs** [comment]
Yeah. I've not been able to make Wireshark work yet. I have a permission denied issue when I try to listen to the P2P device, even when running it with sudo :/

**Comment by benzea** [comment]
Turns out that newer NetworkManager set a default `nm-shared` firewall zone. And this zone blocks access on the relevant port, breaking gnome-network-displays.

See https://gitlab.gnome.org/GNOME/gnome-network-displays/-/issues/134

**Comment by Lctrs** [comment]
Okay, still doesn't work for me. But I feel like we're moving in the right direction here, errors appear further in the process.

**Comment by benzea** [comment]
> Okay, still doesn't work for me. But I feel like we're moving in the right direction here, errors appear further in the process.

Yeah. It seems like we are now dealing with some pipewire bugs. Code to do cleanups was added which is now causing things to break …

**Comment by nedrichards** [comment]
bot, build org.gnome.NetworkDisplays

**Comment by benzea** [comment]
Sooo … the pipewire problems are fixed now upstream (as of https://github.com/PipeWire/pipewire/commit/95192b21f152ba6fa769049a2298aba1221f794d). However, for gnome-network-displays to start working again, both the client library and the pipewire server on the host machine will need to be updated.

I suspect these fixes will take a while to trickle through everywhere.

**Comment by nedrichards** [comment]
Right, does this go on the shelf for a while then? Certainly you'll need a comment in the description that this needs a version of pipewire newer than 0.3.6 on the host to run.

**Comment by Lctrs** [comment]
I tested with this build and a patched version of PipeWire with the needed fixes on the host and streaming to the dummy sink worked well o/

Still can't connect to my TV though, still got that `That device is not up` error with `tcpdump` when I try to debug things, even with the patches you introduced in `wpa_supplicant` in Fedora @benzea

**Comment by Lctrs** [comment]
Deleted my fork by mistake, so I opened a new PR #1677 which replaces this one.

**Pull request closed without merging**